### ISSUE TITLE
[Errno 13] Permission denied

### ISSUE THREAD

**Issue by KoningCheems** [RosettaCommons/RFdiffusion#196]
Hi I am encountering the following error while running RFddifusion inside a Docker Container on HTCondor but I am getting this error. Somehow HYDRA ignores this part : hydra.output_subdir=my_hydra

**Here is my script:**
```
#!/bin/bash

mkdir inputs outputs models my_hydra
mv insulin_target.pdb ./inputs
bash /app/RFdiffusion/scripts/download_models.sh ./models

python3.9 /app/RFdiffusion/scripts/run_inference.py \
	inference.output_prefix=outputs/design_ppi \
	inference.input_pdb=inputs/insulin_target.pdb \
	'contigmap.contigs=[A1-150/0 70-100]' \
	'ppi.hotspot_res=[A59,A83,A91]' \
	inference.num_designs=1 \
	denoiser.noise_scale_ca=0 \
	denoiser.noise_scale_frame=0 \
	inference.model_directory_path=./models \
	hydra.output_subdir=my_hydra

```

**The error:**
```
Error executing job with overrides: ['inference.output_prefix=outputs/design_ppi', 'inference.input_pdb=inputs/insulin_target.pdb', 'contigmap.contigs=[A1-150/0 70-100]', 'ppi.hotspot_res=[A59,A83,A91]', 'inference.num_designs=1', 'denoiser.noise_scale_ca=0', 'denoiser.noise_scale_frame=0', 'inference.model_directory_path=./models']
Traceback (most recent call last):
  File "/app/RFdiffusion/scripts/run_inference.py", line 54, in main
    sampler = iu.sampler_selector(conf)
  File "/usr/local/lib/python3.9/dist-packages/rfdiffusion/inference/utils.py", line 511, in sampler_selector
    sampler = model_runners.SelfConditioning(conf)
  File "/usr/local/lib/python3.9/dist-packages/rfdiffusion/inference/model_runners.py", line 37, in __init__
    self.initialize(conf)
  File "/usr/local/lib/python3.9/dist-packages/rfdiffusion/inference/model_runners.py", line 129, in initialize
    os.mkdir(schedule_directory)
PermissionError: [Errno 13] Permission denied: '/usr/local/lib/python3.9/dist-packages/rfdiffusion/inference/../../schedules'

Set the environment variable HYDRA_FULL_ERROR=1 for a complete stack trace.
```

**Comment by matteoferla** [comment]
That error is due to the fact that the default hydra config is set up as if one had install RFdiffusion via `pip install -e .`.
As a result you need to override `inference.output_prefix` in addition to the `inference.model_directory_path`.

(For non-docker installation, saner solution I found was to copy the config folder into a handy place eg. `$HOME/.cache/RFdiffusion_config`, set a variable for it (`conda env config vars set RFDIFFUSSION_CONFIG=$HOME2/.cache/RFdiffusion_config/inference`) and edited the `base.yaml` file so output_prefix and model_path are set so one just needs to call `run_inference.py --config-path=$RFDIFFUSSION_CONFIG`)

**Comment by aowen-uwmad** [comment]
The issue is evident from these lines in the stack trace:

```
File "/usr/local/lib/python3.9/dist-packages/rfdiffusion/inference/model_runners.py", line 129, in initialize
    os.mkdir(schedule_directory)
PermissionError: [Errno 13] Permission denied: '/usr/local/lib/python3.9/dist-packages/rfdiffusion/inference/../../schedules'
```

The `inference` package, specifically the `model_runners.py` script, is attempting to create the `schedules` directory. A quick peak into the python script shows the [following code](https://github.com/RosettaCommons/RFdiffusion/blob/820bfdfaded8c260b962dc40a3171eae316b6ce0/rfdiffusion/inference/model_runners.py#L122-L125):

```
        if conf.inference.schedule_directory_path is not None:
            schedule_directory = conf.inference.schedule_directory_path
        else:
            schedule_directory = f"{SCRIPT_DIR}/../../schedules"
```

where `SCRIPT_DIR=os.path.dirname(os.path.realpath(__file__))` is the location of `model_runners.py`.
But since this location is in the read-only part of the container, this is an invalid path for creating the `schedules` directory.

TL;DR: add the option `inference.schedule_directory_path = <WRITEABLE_PATH>`. In your case `./schedules` should work.
